lib/vm/src/memory.rs

```diff
@@ -341,6 +374,15 @@ impl VMSharedMemory {
     ) -> Result<Self, MemoryError> {
         Ok(VMOwnedMemory::from_definition(memory, style, vm_memory_location)?.to_shared())
     }
+
+    /// Copies this memory to a new memory
+    pub fn fork(&mut self) -> Result<Self, MemoryError> {
```

Comment:
Let's use clone. I think is a more accurate naming

Reply:
Renamed to `duplicate()`.